Implementing Kafka Consumer to save projects

V3rgi9ly opened the pull request:
1)Added model Project
2)Added ProjectMapper for mapping model Project and ProjectCreatedEvent
3)Added in class ProfileService metho createProject() for save project in database
4)Added in class ProjectCreatedConsumer output message from producer
5)Added file .http for test requests
6)Added sql file for create table Project

V3rgi9ly committed:
1)Added model Project
2)Added ProjectMapper for mapping model Project and ProjectCreatedEvent
3)Added in class ProfileService metho createProject() for save project in database
4)Added in class ProjectCreatedConsumer output message from producer
5)Added file .http for test requests
6)Added sql file for create table Project

DarkRubin commented on src/main/java/com/itmentorcommunityplatform/profileservice/mapper/ProjectMapper.java:
```diff
@@ -0,0 +1,13 @@
+@Mapper(componentModel = "spring")
+public interface ProjectMapper {
+
+    Project toEvent(ProjectCreatedEvent projectCreatedEvent);
```
Метод называется `toEvent`, хотя на самом деле мапит с события в сущность

DarkRubin commented on src/main/java/com/itmentorcommunityplatform/profileservice/service/ProfileService.java:
```diff
@@ -194,4 +200,10 @@ public ProfileResponseDto getProfileByGitHubUrl(String gitHubUrl) {
 
 
+    @Transactional
+    public void createdProject(ProjectCreatedEvent projectCreated){
```
Этот метод явно не относиться к ответственностям сервиса профилей, стояло создать отдельный класс для этого, например `ProjectService`

DarkRubin commented on src/main/java/com/itmentorcommunityplatform/profileservice/consumer/ProjectCreatedConsumer.java:
```diff
@@ -1,16 +1,33 @@
-    public void consumeProjectCreatedEvent(ProjectCreatedEvent projectCreatedEvent) {
-        log.info("Added new project: {}", projectCreatedEvent);
+    public void consumeProjectCreatedEvent(@Payload ProjectCreatedEvent projectCreatedEvent) {
+        log.info("author_telegram_user_id: {}," +
```
Кроме вывода самих полей лучше также стоило бы указать что это мы вообще выводим, чтобы потом можно было найти поиском по логам

DarkRubin commented on src/main/resources/db/changelog/changesets/003-create-table-project.sql:
```diff
@@ -0,0 +1,12 @@
+
+DROP table project;
+
+CREATE TABLE project
```
Не нужно пушить в репозиторий неудачные миграции, их стоило удалить очистить локальную историю миграций, и запушить в репозиторий только финальную версию

DarkRubin commented on src/main/resources/db/changelog/db.changelog-master.yaml:
```diff
@@ -1,3 +1,9 @@
+  - include:
+      file: db/changelog/changesets/003-create-tables.sql
+  - include:
+      file: db/changelog/changesets/004-create-tables.sql
```
И называй файлы миграций более конкретно, чтобы по названию было понятно что в них сделано. Пример можешь посмотреть в auth-service

DarkRubin commented on build.gradle:
```diff
@@ -40,6 +40,10 @@ dependencies {
 	compileOnly 'org.projectlombok:lombok'
 	annotationProcessor 'org.projectlombok:lombok'
 
+	// Mapping
```
Старайся придерживаться существующих правил, если все остальные метки написаны с маленькой буквы, то новую лучше тоже создавать с маленькой

V3rgi9ly committed:
Merge branch 'dev' into feature-save-project-27

V3rgi9ly committed:
Two Iterations
1)Changed in class ProjectCreatedConsumer log for get consumer
2)Deleted from class ProfileService method createProject and created class ProjectSevice for create project
3)Changed name method mapper
4)Deleted sql files
